doc/cephadm/services/monitoring.rst

```diff
@@ -158,6 +158,57 @@ example spec file:
       port: 4200
       protocol: http
 
+.. _cephadm_default_images:
+
+Default images
+~~~~~~~~~~~~~~
+
+``cephadm`` stores a local copy of the ``cephadm`` binary in
+``var/lib/ceph/{FSID}/cephadm.{DIGEST}``, where ``{DIGEST}`` is an alphanumeric
+string representing the currently-running version of Ceph. In the Squid and
+Reef releases of Ceph, this ``cephadm`` file is stored as a zip file and must
+be unzipped before its contents can be examined.
+
+This procedure explains how to generate a list of the default container images
+used by ``cephadm``.
+
+.. note:: This procedure applies only to the Reef and Squid releases of Ceph.
+   If you are using a different version of Ceph, you cannot use this procedure
+   to examine the list of default containers used by cephadm. Make sure that
+   you are reading the documentation for the release of Ceph that you have
+   installed.
+
+#. To create a directory called ``cephadm_dir``, run the following command:
+
+   .. prompt:: bash #
+
+      mkdir cephadm_dir
+
+#. To unzip ``/var/lib/ceph/{FSID}/cephadm.{DIGEST}`` in the directory
+   ``cephadm_dir``, run a command of the following form:
+
+   .. prompt:: bash #
+
+      unzip /var/lib/ceph/{FSID}/cephadm.{DIGEST} -d cephadm_dir > /dev/null
+
+   ::
+
+      warning [/var/lib/ceph/{FSID}/cephadm.{DIGEST}]:  14 extra bytes at
+      beginning or within zipfile
+      (attempting to process anyway)
+
+
+#. To use ``egrep`` to search for the string ``_IMAGE`` in the file
+   ``cephadm_dir/__main__.py``, run the following command: 
+
+   .. prompt:: bash #
+
+      egrep "_IMAGE" cephadm_dir/cephadmlib/__main__.py
+
+   ::
+
+      DEFAULT_IMAGE = 'quay.ceph.io/ceph-ci/ceph:main'
```

Comment:
Not sure if we intended for this to show the actual set of default images on reef or not, but I just ran this on a reef cephadm zip and got (had to make it search of `IMAGE =` instead to filter some other results)
```
[root@vm-00 ~]# grep -E "_IMAGE =" cephadm_dir/__main__.py
DEFAULT_IMAGE = 'quay.io/ceph/ceph:v18'
DEFAULT_PROMETHEUS_IMAGE = 'quay.io/prometheus/prometheus:v2.43.0'
DEFAULT_LOKI_IMAGE = 'docker.io/grafana/loki:2.4.0'
DEFAULT_PROMTAIL_IMAGE = 'docker.io/grafana/promtail:2.4.0'
DEFAULT_NODE_EXPORTER_IMAGE = 'quay.io/prometheus/node-exporter:v1.5.0'
DEFAULT_ALERT_MANAGER_IMAGE = 'quay.io/prometheus/alertmanager:v0.25.0'
DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/ceph-grafana:9.4.7'
DEFAULT_HAPROXY_IMAGE = 'quay.io/ceph/haproxy:2.3'
DEFAULT_KEEPALIVED_IMAGE = 'quay.io/ceph/keepalived:2.2.4'
DEFAULT_NVMEOF_IMAGE = 'quay.io/ceph/nvmeof:1.0.0'
DEFAULT_SNMP_GATEWAY_IMAGE = 'docker.io/maxwo/snmp-notifier:v1.2.1'
DEFAULT_ELASTICSEARCH_IMAGE = 'quay.io/omrizeneva/elasticsearch:6.8.23'
DEFAULT_JAEGER_COLLECTOR_IMAGE = 'quay.io/jaegertracing/jaeger-collector:1.29'
DEFAULT_JAEGER_AGENT_IMAGE = 'quay.io/jaegertracing/jaeger-agent:1.29'
DEFAULT_JAEGER_QUERY_IMAGE = 'quay.io/jaegertracing/jaeger-query:1.29'
```
what you have here is what you'd get if you ran the same against main

Reply:
Actually, just realized your patch says to check `cephadm_dir/cephadmlib/__main__.py` rather than `cephadm_dir/__main__.py` so we'll need to adjust that @zdover23

Reply:
I'll add this equals sign in. (It's going to take a little while, because I have another branch open at the moment.)

Reply:
I've added in the equals sign now.

Reply:
> Actually, just realized your patch says to check `cephadm_dir/cephadmlib/__main__.py` rather than `cephadm_dir/__main__.py` so we'll need to adjust that @zdover23

Good catch. I've corrected it.

Reply:
> Not sure if we intended for this to show the actual set of default images on reef or not, but I just ran this on a reef cephadm zip and got (had to make it search of `IMAGE =` instead to filter some other results)
> 
> ```
> [root@vm-00 ~]# grep -E "_IMAGE =" cephadm_dir/__main__.py
> DEFAULT_IMAGE = 'quay.io/ceph/ceph:v18'
> DEFAULT_PROMETHEUS_IMAGE = 'quay.io/prometheus/prometheus:v2.43.0'
> DEFAULT_LOKI_IMAGE = 'docker.io/grafana/loki:2.4.0'
> DEFAULT_PROMTAIL_IMAGE = 'docker.io/grafana/promtail:2.4.0'
> DEFAULT_NODE_EXPORTER_IMAGE = 'quay.io/prometheus/node-exporter:v1.5.0'
> DEFAULT_ALERT_MANAGER_IMAGE = 'quay.io/prometheus/alertmanager:v0.25.0'
> DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/ceph-grafana:9.4.7'
> DEFAULT_HAPROXY_IMAGE = 'quay.io/ceph/haproxy:2.3'
> DEFAULT_KEEPALIVED_IMAGE = 'quay.io/ceph/keepalived:2.2.4'
> DEFAULT_NVMEOF_IMAGE = 'quay.io/ceph/nvmeof:1.0.0'
> DEFAULT_SNMP_GATEWAY_IMAGE = 'docker.io/maxwo/snmp-notifier:v1.2.1'
> DEFAULT_ELASTICSEARCH_IMAGE = 'quay.io/omrizeneva/elasticsearch:6.8.23'
> DEFAULT_JAEGER_COLLECTOR_IMAGE = 'quay.io/jaegertracing/jaeger-collector:1.29'
> DEFAULT_JAEGER_AGENT_IMAGE = 'quay.io/jaegertracing/jaeger-agent:1.29'
> DEFAULT_JAEGER_QUERY_IMAGE = 'quay.io/jaegertracing/jaeger-query:1.29'
> ```
> 
> what you have here is what you'd get if you ran the same against main

I've replaced what was there originally with the following: 

```
> [root@vm-00 ~]# grep -E "_IMAGE =" cephadm_dir/__main__.py
> DEFAULT_IMAGE = 'quay.io/ceph/ceph:v18'
> DEFAULT_PROMETHEUS_IMAGE = 'quay.io/prometheus/prometheus:v2.43.0'
> DEFAULT_LOKI_IMAGE = 'docker.io/grafana/loki:2.4.0'
> DEFAULT_PROMTAIL_IMAGE = 'docker.io/grafana/promtail:2.4.0'
> DEFAULT_NODE_EXPORTER_IMAGE = 'quay.io/prometheus/node-exporter:v1.5.0'
> DEFAULT_ALERT_MANAGER_IMAGE = 'quay.io/prometheus/alertmanager:v0.25.0'
> DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/ceph-grafana:9.4.7'
> DEFAULT_HAPROXY_IMAGE = 'quay.io/ceph/haproxy:2.3'
> DEFAULT_KEEPALIVED_IMAGE = 'quay.io/ceph/keepalived:2.2.4'
> DEFAULT_NVMEOF_IMAGE = 'quay.io/ceph/nvmeof:1.0.0'
> DEFAULT_SNMP_GATEWAY_IMAGE = 'docker.io/maxwo/snmp-notifier:v1.2.1'
> DEFAULT_ELASTICSEARCH_IMAGE = 'quay.io/omrizeneva/elasticsearch:6.8.23'
> DEFAULT_JAEGER_COLLECTOR_IMAGE = 'quay.io/jaegertracing/jaeger-collector:1.29'
> DEFAULT_JAEGER_AGENT_IMAGE = 'quay.io/jaegertracing/jaeger-agent:1.29'
> DEFAULT_JAEGER_QUERY_IMAGE = 'quay.io/jaegertracing/jaeger-query:1.29'
```